core/shared/src/main/scala/scalaz/zio/duration/Duration.scala

```diff
@@ -135,6 +135,10 @@ object Duration {
     case _                      => Infinity
   }
 
+  final def fromJava(duration: JavaDuration): Duration =
+    if (duration.getSeconds == Long.MaxValue) Infinity
+    else fromNanos(duration.toNanos)
```

Comment:
It will throw for any Duration that is created with `Duration.ofDays(x)` when `x > 365*293`.
Could you provide implementation that wouldn't throw for long durations (especially ones created with `.ofDays`)?

Reply:
> It will throw for any Duration that is created with `Duration.ofDays(x)` when `x > 365*293`.
> Could you provide implementation that wouldn't throw for long durations (especially ones created with `.ofDays`)?

Yep, will do. I think I misunderstood the existing ZIO implementation, which is that *all* durations longer than `Long.MaxValue` nanoseconds are counted as `Infinity`. This solves the question I had about which Java durations should be converted to a ZIO infinity. I will push a fix shortly.